client/my-sites/plugins/plugins-browser/single-list-view/index.jsx

```diff
@@ -44,7 +44,7 @@ const SingleListView = ( { category, plugins, isFetching, siteSlug, sites, noHea
 	const { localizePath } = useLocalizedPlugins();
 
 	const installedPlugins = useSelector( ( state ) =>
-		getPlugins( state, siteObjectsToSiteIds( sites ) )
+		siteId ? getPlugins( state, siteObjectsToSiteIds( sites ) ) : []
```

Comment:
Nit: I would do this verification down below instead of changing the `installedPlugins` value, if we need to use this variable in the future it might not be clear why its empty in some cases.

```js
.filter( ( plugin ) => !siteId || isNotInstalled( plugin, installedPlugins ) );
```

Reply:
I have considered this too but thought that avoiding `getPlugins( state, siteObjectsToSiteIds( sites ) ) ` when not needed would save some cpu time. Looking at your comment though made me realise that filtering there could be improved as I did in https://github.com/Automattic/wp-calypso/pull/92515/commits/620f178ca69c4f86d8f40e14acaa6f923f38d41c by bundling the filtering and avoiding a call to `isNotInstalled`.